src/client.rs

```diff
@@ -125,6 +127,22 @@ impl ImageLayer {
     }
 }
 
+/// Implementing Hash to enable determining uniqueness
+/// eg via https://docs.rs/itertools/latest/itertools/structs/struct.Unique.html
+impl Hash for ImageLayer {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.data.hash(state);
+        self.media_type.hash(state);
+        if let Some(annotations) = self.annotations.clone() {
```

Comment:
I think I'd rather swap annotations to be a `BTreeMap` and break the API a little bit. It makes sense that we'd want these things to be hashable and comparable and the `BTreeMap` is already `Hash` and `Eq`. That way we don't need another dependency and the traits can be entirely derived

Reply:
👍 Nice, much simpler.  I left the unit test in...

Reply:
Yeah I'm totally fine with itertools as a dev dep